testsuite/unittests/test_connect.sli

```diff
@@ -180,49 +180,38 @@ ResetKernel
 
 (Running Connect tests with illegal combinations) =
 ResetKernel
-/generator /poisson_generator Create def
-/multimeter /multimeter Create def
-/recordable /iaf_psc_alpha Create def
 
+/n_connections 0 def
+modeldict
 {
-  generator generator Connect
-} fail_or_die
+  pop  % remove model id
+  /model_a Set
+  /type_a model_a GetDefaults /element_type get def 
+  /m_a model_a Create def
```

Comment:
It looks to me as if you do not need `type_a` anywhere, so I think you can shorten this to
```
pop exch Create /node_a Set
```
and similarly for `b` below.